<launch>

	<!-- Change ROS logger levels -->
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find dolphin_slam)/config/custom_rosconsole.conf"/>

	<!-- Name of bag file -->
	<arg name="bagfile" default="$(find dolphin_slam)/experiments/uwsim/test/test.bag"/>

	<!-- Vocabulary path -->
    <arg name="vocabulary_path" value ="$(find dolphin_slam)/experiments/uwsim/training/"/>

	<!-- Play messages from bagfile -->
    <node pkg="rosbag" type="play" name="play" args="$(arg bagfile) -q -r 3 --clock" required="false"/>
	
	<!-- process images to extract surf keypoints -->
    <node pkg="dolphin_slam" type="image_processing_node" name="image_processing_node" required="true">
       	<param name="surf_threshold" type="int" value="100" />		
		<param name="image_topic" type="string" value="/uwsim/camera1" />		
		<param name="image_transport" type="string" value="compressed" />		
		<param name="descriptors_topic" type="string" value="/descriptors" />		
		<param name="keypoints_topic" type="string" value="/image_keypoints" />
        <param name="frames_to_jump" type="int" value="60" />
        <param name="use_selected_images" type="bool" value="false" />
    </node>

	<!-- view surf keypoints plotted on image  -->
    <node pkg="image_view" type="image_view" name="keypoints_view" args="image:=/image_keypoints" respawn="false"/>

	<!-- train fabmap algorithm  -->
    <node pkg="dolphin_slam" type="fabmap_training" name="fabmap_training" required="true">
		<param name="descriptors_topic" type="string" value="/descriptors" />		
        <param name="fabmap_vocab" type="string" value="$(arg vocabulary_path)fabmap_vocab.xml" />
        <param name="fabmap_descriptors" type="string" value="$(arg vocabulary_path)fabmap_descriptors.xml"/>
        <param name="fabmap_tree" type="string" value="$(arg vocabulary_path)fabmap_tree.xml" />
        <param name="cluster_size" type="double" value="0.6" />
        <param name="cluster_count" type="int" value="500" />
        <param name="clustering_algorithm" type="string" value="kmeans" />        
        <param name="fabmap_implementation" type="string" value="original" />        
        <param name="dataset_name" type="string" value="uwsim" />
        <param name="dataset_path" type="string" value="$(arg vocabulary_path)" />
    </node>

	<!-- record topic messages on a bag file  -->
    <!-- node pkg="rosbag" type="record" name="record" args="\descriptors \image_keypoints -o fabmap_training" required="false"/ -->
	
</launch>

